From: Ilya Dryomov <idryomov@gmail.com>
Date: Fri, 12 Jan 2018 17:22:10 +0100
Subject: rbd: set max_segment_size to UINT_MAX
Git-commit: 24f1df60ce943aee107b3cb99b37a0152c9dd47a
Patch-mainline: v4.17-rc1
References: FATE#324714

Commit 21acdf45f495 ("rbd: set max_segments to USHRT_MAX") removed the
limit on max_segments.  Remove the limit on max_segment_size as well.

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Reviewed-by: Alex Elder <elder@linaro.org>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 drivers/block/rbd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/block/rbd.c
+++ b/drivers/block/rbd.c
@@ -4369,7 +4369,7 @@ static int rbd_init_disk(struct rbd_devi
 	blk_queue_max_hw_sectors(q, segment_size / SECTOR_SIZE);
 	q->limits.max_sectors = queue_max_hw_sectors(q);
 	blk_queue_max_segments(q, USHRT_MAX);
-	blk_queue_max_segment_size(q, segment_size);
+	blk_queue_max_segment_size(q, UINT_MAX);
 	blk_queue_io_min(q, segment_size);
 	blk_queue_io_opt(q, segment_size);
 
